src/test/shared/treeview/resourceTreeDataProvider.test.ts

```diff
@@ -100,6 +100,20 @@ describe('ResourceTreeDataProvider', function () {
         assert.deepStrictEqual(newItem0, item0)
     })
 
+    it('can refresh a node independently of the rest of the tree', async function () {
+        const provider = new ResourceTreeDataProvider(getRoot(flatTree))
+        const [node0, node1] = await provider.getChildren()
+        const item0 = await provider.getTreeItem(node0)
+        const item1 = await provider.getTreeItem(node1)
+        provider.refresh(node1)
+
+        const newItem0 = await provider.getTreeItem(node0)
+        const newItem1 = await provider.getTreeItem(node1)
+        assert.strictEqual(newItem0, item0)
+        assert.notStrictEqual(newItem1, item1)
+        assert.deepStrictEqual(newItem1, item1)
```

Comment:
The same args are "not strict equal", but _are_ "deep strict equal"? This may be worth a 1-line explaining the subtlety.

Reply:
They have the same data but are not the same references.
It's the most straightforward way to explicitly test caching behavior albeit somewhat abstract.